src/plugins/intel_cpu/tests/unit/cpu_tensor_test.cpp

```diff
@@ -0,0 +1,98 @@
+// Copyright (C) 2018-2023 Intel Corporation
+// SPDX-License-Identifier: Apache-2.0
+//
+
+#include <gtest/gtest.h>
+
+#include <openvino/core/shape.hpp>
+#include <openvino/core/strides.hpp>
+#include <openvino/core/type/element_type.hpp>
+#include "openvino/core/except.hpp"
+#include "openvino/core/partial_shape.hpp"
+
+#include "cpu_memory.h"
+#include "cpu_tensor.h"
+
+using namespace ov::intel_cpu;
+using namespace InferenceEngine;
+
+using CPUTensorTest = ::testing::Test;
+
+inline ov::Strides byteStrides(const ov::Strides& strides, const ov::element::Type& type) {
+    ov::Strides byte_strides(strides.size());
+    for (size_t i = 0; i < strides.size(); ++i)
+        byte_strides[i] = strides[i] * type.size();
+    return byte_strides;
+}
+
+inline MemoryPtr create_memory(Precision prc, const Shape& shape) {
+    dnnl::engine eng(dnnl::engine::kind::cpu, 0);
+    CpuBlockedMemoryDescPtr desc;
+    desc = std::make_shared<CpuBlockedMemoryDesc>(prc, shape);
+    return std::make_shared<Memory>(eng, desc);
+}
+
+TEST_F(CPUTensorTest, canCreateTensor) {
+    Shape shape{4, 3, 2};
+    ov::Shape ov_shape = shape.toPartialShape().to_shape();
+
+    ov::intel_cpu::Tensor t(create_memory(Precision::FP32, shape));
+    const std::size_t totalSize = ov::shape_size(ov_shape);
+    ASSERT_EQ(totalSize, t.get_size());
+    ASSERT_NE(nullptr, t.data());
+    ASSERT_EQ(ov::element::f32, t.get_element_type());
+    ASSERT_EQ(ov_shape, t.get_shape());
+    ASSERT_NE(ov_shape, t.get_strides());
+    ASSERT_EQ(byteStrides(ov::Strides({6, 2, 1}), t.get_element_type()), t.get_strides());
+    ASSERT_EQ(ov::element::f32.size() * totalSize, t.get_byte_size());
+    ASSERT_THROW(t.data(ov::element::i64), ov::Exception);
+    ASSERT_THROW(t.data<std::int32_t>(), ov::Exception);
+}
+
+TEST_F(CPUTensorTest, canAccessF16Tensor) {
+    Shape shape = {4, 3, 2};
+    ov::intel_cpu::Tensor t(create_memory(Precision::FP16, shape));
+    EXPECT_NE(nullptr, t.data());
+    ASSERT_EQ(ov::element::f16, t.get_element_type());
+    EXPECT_NO_THROW(t.data(ov::element::f16));
+    EXPECT_NO_THROW(t.data<ov::float16>());
+    EXPECT_THROW(t.data<ov::bfloat16>(), ov::Exception);
+    EXPECT_THROW(t.data<std::uint16_t>(), ov::Exception);
+    EXPECT_THROW(t.data<std::int16_t>(), ov::Exception);
+}
+
+TEST_F(CPUTensorTest, emptySize) {
+    ov::intel_cpu::Tensor t(create_memory(Precision::FP32, {0}));
+    ASSERT_NE(nullptr, t.data());
```

Comment:
This check would fail, as MemoryMngrWithReuse would never return nullptr even if required size is zero. We might force mem manager return nullptr if input size is zero. How do you think? @maxnick

Reply:
AFAIK, GetData() should return nulltpr if the initial memory size is zero and the memory has not been yet resized. https://github.com/maxnick/openvino/blob/dae9ac1d6282a83a20036379fe7d57c1572ab79b/src/plugins/intel_cpu/src/cpu_memory.cpp#L193-L200

Reply:
my mistake. It is checking "ASSERT_NE" here. That means ov core expects tensor.data() returns a non-null ptr even if it is an empty tensor.
Sounds we need resize the tensor. But it still fails because of if (size > _memUpperBound) check in MemoryMngrWithReuse::resize. So I did a slight change in resize. Would you please comment?

Reply:
done